src/vulkan/engine_vulkan.cc

```diff
@@ -32,27 +31,32 @@ namespace {
 
 const FormatType kDefaultFramebufferFormat = FormatType::kB8G8R8A8_UNORM;
 
-VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
+Result ToVkShaderStage(ShaderType type, VkShaderStageFlagBits* ret) {
   switch (type) {
     case kShaderTypeGeometry:
-      return VK_SHADER_STAGE_GEOMETRY_BIT;
+      *ret = VK_SHADER_STAGE_GEOMETRY_BIT;
+      break;
     case kShaderTypeFragment:
-      return VK_SHADER_STAGE_FRAGMENT_BIT;
+      *ret = VK_SHADER_STAGE_FRAGMENT_BIT;
+      break;
     case kShaderTypeVertex:
-      return VK_SHADER_STAGE_VERTEX_BIT;
+      *ret = VK_SHADER_STAGE_VERTEX_BIT;
+      break;
     case kShaderTypeTessellationControl:
-      return VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT;
+      *ret = VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT;
+      break;
     case kShaderTypeTessellationEvaluation:
-      return VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT;
+      *ret = VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT;
+      break;
     case kShaderTypeCompute:
-      return VK_SHADER_STAGE_COMPUTE_BIT;
-    case kShaderTypeMulti:
-      // It's an error if this arrives here ...
+      *ret = VK_SHADER_STAGE_COMPUTE_BIT;
       break;
+    case kShaderTypeMulti:
```

Comment:
what the heck is multi?   the header file isn't very illuminating.  But that's for another CL

Reply:
It's a spirv module with multiple shaders in it.So you can have a vertex and fragment shaders in a single module. so, a multi shader module, heh